src/js/component/reportForm/index.js

```diff
@@ -0,0 +1,168 @@
+      selectedMonth,
+    };
+    console.log('formData', formData);
+
+    //TODO- replace with actual endpoint
+
+    // post('url', formData)
+    //   .then((res) => {})
+    //   .catch((error) => {
+    //     const obj = {
+    //       title: 'error',
+    //       message: error.message,
+    //       status: true,
+    //       type: 'error',
+    //     };
+    //     dispatch(manageToast(obj));
+    //     dispatch(setOverlayStatus(false));
+    //   });
+  };
+  return (
+    <form onSubmit={handleSubmit} className={classes.root}>
+      <FormControl sx={{ marginBottom: '10px' }}>
+        <FormLabel id='demo-controlled-radio-buttons-group'>
+          Pick the format of report:{' '}
+        </FormLabel>
+        <RadioGroup
+          aria-labelledby='demo-controlled-radio-buttons-group'
+          name='controlled-radio-buttons-group'
+          value={reportFormat}
+          onChange={(e) => setReportFormat(e.target.value)}
+        >
+          <FormControlLabel value='daily' control={<Radio />} label='Daily' />
+          <FormControlLabel
+            value='monthly'
+            control={<Radio />}
+            label='Monthly'
+          />
+        </RadioGroup>
+      </FormControl>
+      {reportFormat === 'daily' && (
+        <>
+          <FormControl sx={{ marginBottom: '10px', width: '100%' }}>
+            <LocalizationProvider dateAdapter={AdapterDayjs}>
+              <DatePicker
+                label='Pick date'
+                value={selectedDate}
+                onChange={(newValue) => setSelectedDate(newValue)}
+                disableFuture={true}
+              />
+            </LocalizationProvider>
+          </FormControl>
+        </>
+      )}
+
+      {reportFormat === 'monthly' && (
+        <>
+          <FormControl
+            sx={{ marginBottom: '10px' }}
+            required={reportFormat === 'monthly'}
+          >
+            <InputLabel id='demo-simple-select-label'>Month</InputLabel>
+            <Select
+              labelId='demo-simple-select-label'
+              id='demo-simple-select'
+              value={selectedMonth}
+              label='Month'
+              onChange={(e) => setSelectedMonth(e.target.value)}
+            >
+              {months?.map((month) => (
+                <MenuItem value={month}>{month}</MenuItem>
+              ))}
+            </Select>
+          </FormControl>
+
+          <FormControl
+            fullWidth
+            sx={{ marginBottom: '10px' }}
+            required={reportFormat === 'monthly'}
+          >
+            <InputLabel id='demo-simple-select-label'>Year</InputLabel>
+            <Select
+              labelId='demo-simple-select-label'
+              id='demo-simple-select'
+              value={selectedYear}
+              label='Year'
+              onChange={(e) => setSelectedYear(e.target.value)}
+            >
+              {years?.map((year) => (
+                <MenuItem value={year}>{year}</MenuItem>
```

Comment:
key missing

Reply:
whenever you use map, always remember to put key..and do not put index as it value..